contracts/0.4.24/StETHPermit.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
```

Comment:
Not sure that we can easily recopyright OpenZeppelin code like this.

Reply:
For the copyright: you are totally right, thank you!

---

Regarding the separate contract:

it would require exposing `_approve` function for it with additional permissions checks, and OZ implementation has its own storage for the whole token inside (ERC20Permit is derived from ERC20)
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Permit.sol